metrics/stackdriver_exporter.go

```diff
@@ -80,6 +90,41 @@ func newStackdriverExporter(config *metricsConfig, logger *zap.SugaredLogger) (v
 	return e, nil
 }
 
+// getStackdriverExporterClientOptions creates client options for the opencensus Stackdriver exporter from the given stackdriverClientConfig.
+// On error, an empty array of client options is returned.
+func getStackdriverExporterClientOptions(sdconfig *stackdriverClientConfig) ([]option.ClientOption, error) {
+	var co []option.ClientOption
+	if sdconfig.GCPSecretName != "" {
+		secret, err := getStackdriverSecret(sdconfig)
+		if err != nil {
+			return co, err
+		}
+
+		co = []option.ClientOption{convertSecretToExporterOption(secret)}
+	}
+
+	return co, nil
+}
+
+// getGCPMetadata returns GCP metadata required to export metrics
+// to Stackdriver. Values explicitly set in the config take the highest precedent.
+func getGCPMetadata(config *metricsConfig) *gcpMetadata {
```

Comment:
As I mentioned above, both project and location can be defaulted if they are empty strings by the library already:

```golang
    // Location is the identifier of the GCP or AWS cloud region/zone in which
    // the data for a resource is stored.
    // If not set, it will default to the location provided by the metadata server.
    //
    // It will be used in the location label of a Stackdriver monitored resource
    // if the resource does not inherently belong to a specific project, e.g.
    // on-premise resource like k8s_container or generic_task.
```